### ISSUE TITLE
Improved string interpolation lexing removes start of string interpolations with parenthesis

### ISSUE THREAD

**Issue by bauglir** [JuliaLang/Tokenize.jl#175]
The improved string interpolation lexing released in [v0.5.14](https://github.com/JuliaLang/Tokenize.jl/releases/tag/v0.5.14) (i.e. #173) swallows the start of string interpolations with parenthesis.

```julia
(@v1.6) pkg> add Tokenize@0.5.14
... snip ...
julia> using Tokenize
julia> collect(tokenize("\"\$(foo)\""))
2-element Vector{Tokenize.Tokens.Token}:
 1,1-1,8          STRING         "foo)\""
 1,9-1,8          ENDMARKER      ""
```

On v0.5.13 the same example produces
```julia
(@v1.6) pkg> add Tokenize@0.5.13
... snip ...
julia> using Tokenize
julia> collect(tokenize("\"\$(foo)\""))
2-element Vector{Tokenize.Tokens.Token}:
 1,1-1,8          STRING         "\"$(foo)\""
 1,9-1,8          ENDMARKER      ""
```

This breaks JuliaFormatter.jl when formatting some of my codebases. I'm not sure, but it seems to me the v0.5.13 behavior was the correct behavior.

When the parenthesis are removed the tokenization behaves as expected
```julia
julia> collect(tokenize("\"\$foo\""))
2-element Vector{Tokenize.Tokens.Token}:
 1,1-1,6          STRING         "\"$foo\""
 1,7-1,6          ENDMARKER      ""
```
However, that doesn't work in case more complex expressions are interpolated (e.g. when indexing into a struct).

The examples have been generated using the current `julia:1.6.0-alpine` Docker image:
```julia
julia> versioninfo()
Julia Version 1.6.0-rc3
Commit 23267f0d46 (2021-03-16 17:04 UTC)
Platform Info:
  OS: Linux (x86_64-linux-musl)
  CPU: Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
Environment:
  JULIA_GPG = 3673DF529D9049477F76B37566E3C7DC03D6E495
  JULIA_PATH = /usr/local/julia
  JULIA_VERSION = 1.6.0-rc3
```

But I've also noticed this on the `1.5.3-alpine` and `1.5.3` Docker images and on 1.6.0 on OSX.

**Comment by pfitzseb** [comment]
Sorry about that. Turns out we didn't have any tests for that :) Fixed in #176.

**Comment by bauglir** [comment]
Excellent! Cannot have tests for everything 😀

Thanks for the fast turnaround!
